nifi-nar-bundles/nifi-jwt-bundle/nifi-jwt-processors/src/main/java/org/apache/nifi/processors/jwt/UnsignJWT.java

```diff
@@ -0,0 +1,236 @@
+            .displayName("Approved Public Keys")
+            .description("Path to the directory containing approved public certificates. Certificates are expected to have "
+                + "extension \".pub\" and be x509 PEM files")
+            .required(true).addValidator(CustomValidators.DIRECTORY_HAS_PUBLIC_KEYS_VALIDATOR).build();
+
+    public static final PropertyDescriptor JWT_ATTRIBUTE_NAME = new PropertyDescriptor.Builder().name("JWT_ATTRIBUTE_NAME")
+            .displayName("JWT Attribute name")
+            .description("The name of the attribute which contains the JWT. Leave empty to read JWT from the flowfile content")
+            .required(false).defaultValue("").addValidator(StandardValidators.ATTRIBUTE_KEY_PROPERTY_NAME_VALIDATOR).build();
+
+    public static final Relationship SUCCESS_REL = new Relationship.Builder().name("success")
+            .description("A FlowFile is routed to this relationship after its JWT has been validated and unsigned")
+            .build();
+
+    public static final Relationship FAILURE_REL = new Relationship.Builder().name("failure").description(
+            "A FlowFile is routed to this relationship if the JWT cannot be validated or unsigned for any reason")
+            .build();
+
+    public static final String HEADER_ATTRIBUTE = "jwt.header";
+    public static final String FOOTER_ATTRIBUTE = "jwt.footer";
+    public static final String JWT_PREFIX_ATTRIBUTE = "jwt.";
+
+    public static final String PUBLIC_KEY_PREFIX = "-----BEGIN PUBLIC KEY-----\n";
+    public static final String PUBLIC_KEY_SUFFIX = "-----END PUBLIC KEY-----";
+    public static final String SSH_RSA_PREFIX = "ssh-rsa ";
+
+    private File publicCerts = null;
+    private boolean jwtInBody = true;
+    private String jwtAttribute;
+    private KeyFactory kf;
+
+    private List<PropertyDescriptor> descriptors;
+    private Set<Relationship> relationships;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<PropertyDescriptor>();
+        descriptors.add(PUBLIC_KEYS_PATH);
+        descriptors.add(JWT_ATTRIBUTE_NAME);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(SUCCESS_REL);
+        relationships.add(FAILURE_REL);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        // read public cert from property
+        publicCerts = new File(context.getProperty(PUBLIC_KEYS_PATH).getValue());
+        try {
```

Comment:
If the JVM doesn't have the RSA algorithm available, it should fail and communicate that to the user.